### PR TITLE
Use Vault for HASH AWS credentials

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR switches from our secrets-based hardcoded AWS credentials to using our vault instance to get temporary credentials for pushing images.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1204675418698206/f) _(internal)_

## 🔍 What does this change?

- Changes actions and workflows related to pushing to ECR
- Driveby todo fixes

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

## 🐾 Next steps

- Migrate to STS based credentials in terraform
- Automate Terraform plan/apply

## 🛡 What tests cover this?

- CI/CD

## ❓ How to test this?

Hard to test manually, This run contains the result from this change https://github.com/hashintel/hash/actions/runs/5089339009 (no effective change)

### CODE REVIEW

File: .github/workflows/hash-backend-cd.yml [hunk 1]
```diff
@@ -5,18 +5,15 @@ on:
     branches:
       - main
     paths:
+      - "apps/hash-graph/**"
```
Comment (TimDiekmann) [inline .github/workflows/hash-backend-cd.yml:8]: Do we want to sort them by their name?

File: .github/workflows/hash-backend-cd.yml [hunk 2]
```diff
@@ -45,9 +42,27 @@ jobs:
   build-graph:
     name: Build and push HASH graph image
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
```
Comment (TimDiekmann) [inline .github/workflows/hash-backend-cd.yml:47]: We should probably add `permissions` to all actions at some point.

**Review by TimDiekmann — APPROVED** [review]
CHanges look good to me and seem to work, thank you!

Regarding the failing dependency check: We spoke offline about it and agreed that it's safe to use `v2` as `v2` points to the `v2.5.0` tag and the issue was [fixed in version 2.2](https://github.com/hashicorp/vault-action/blob/v2/CHANGELOG.md#220-may-6th-2021)

File: .github/workflows/hash-backend-cd.yml (hunk 1 above)
Comment (thehabbos007) [reply .github/workflows/hash-backend-cd.yml:8]: Probably yes, but I started out just grouping them by their path at least

File: .github/workflows/hash-backend-cd.yml (hunk 2 above)
Comment (thehabbos007) [reply .github/workflows/hash-backend-cd.yml:47]: Yes we should. These are necessary specifically for the id-token vault auth though

**Pull request merged**